File: app-java/src/main/java/science/monke/internal/workflow/DefaultWorkflow.java
package science.monke.internal.workflow;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Component;
import science.monke.internal.event.Event;

@Component
public class DefaultWorkflow implements Workflow {

  private final Event orderEvent;

  @Autowired
  public DefaultWorkflow(final Event orderEvent) {
    this.orderEvent = orderEvent;
  }

  @Override
  public WorkflowName getName() {
    return WorkflowName.DEFAULT_WORKFLOW;
  }

  @Override
  public WorkflowObject execute(final WorkflowObject workflowObject) {
    return orderEvent.execute(workflowObject);
  }
}

File: app-kotlin/src/main/kotlin/science/monke/internal/workflow/DefaultWorkflow.kt
package science.monke.internal.workflow

import org.springframework.stereotype.Component
import science.monke.internal.event.Event

@Component
class DefaultWorkflow(val orderEvent: Event) : Workflow {

    override suspend fun getName(): WorkflowName {
        return WorkflowName.DEFAULT_WORKFLOW
    }

    override suspend fun execute(workflowObject: WorkflowObject): WorkflowObject {
        return orderEvent.execute(workflowObject)
    }
}
